### DISCUSSION TITLE
TreeView inside '@opensumi/ide-components'?

### DISCUSSION THREAD

**Discussion by mister-teddy** [opensumi/core#1941] [Q&A] [answered]
I wonder if there is any built-in component for building a tree view like this? Or a 3rd-party react library would be needed?

<img width="249" alt="Screenshot 2022-11-08 at 22 41 55" src="https://user-images.githubusercontent.com/29925961/200609733-afd0848b-45b5-442d-9e85-54ee5312fe6c.png">

Thank you 🙏

**Comment by erha19** [comment, marked as answer]
@nguyenhongphat0 Hi, I have created a sample about 'How to use OpenSumi BasicRecycleTree', see https://github.com/opensumi/opensumi-module-samples/pull/13.

The `BasicRecycleTree` provide some simple features to help you build a TreeView, it's very simple, and based on the `RecycleTree` component, you can evaluate whether it meets your needs, at the same time, there is also a more flexible `RecycleTree` component in the `@opensumi/ide-components`, which can basically meet the customization capabilities of all TreeView views, if you expect to use, you can also refer to e.g [opened-editor.tsx#L144](https://github.com/opensumi/core/blob/b8066860c7470467bfb923800c54dddcadcf3d0b/packages/opened-editor/src/browser/opened-editor.tsx#L144).

Hope this can help you ~

**Comment by mister-teddy** [comment]
Thank you very much for your quick response @erha19 🙏
